lib/telemetry_sampler.ex

```diff
@@ -1,5 +1,400 @@
 defmodule Telemetry.Sampler do
   @moduledoc """
-  Allows to periodically gather measurements and publish them as `Telemetry` events
+  Allows to periodically collect measurements and publish them as `Telemetry` events.
+
+  ## Measurement specifications
+
+  Measurement specifications (later "specs") describe how the measurement should be performed and
+  what event should be emitted once the mesured value is collected. There are three possible values
+  for measurement spec:
+
+  1. An MFA tuple returning a single sample or a list of samples. Sample is an opaque data structure
+     which consists of event name, collected measurement value and event metadata. Samples can
+     be constructed using the `sample/3` function. Sampler will dispatch a `Telemetry` event
+     for each sample returned by the MFA invokation.
```

Comment:
Wouldn't it be better to have those functions call telemetry directly then?

Reply:
Hm, probably.. I liked to idea of returning samples, it kind of made sense in my head, but now I see that it might just be unnecessary abstraction. It doesn't hide much information, since we're stating explicitly that the events will be dispatched using `Telemetry` anyway.